processor/src/substrate_signer.rs

```diff
@@ -204,16 +211,19 @@ impl<D: Db> SubstrateSigner<D> {
   }
 
   pub async fn sign(&mut self, txn: &mut D::Transaction<'_>, batch: Batch) {
-    if SubstrateSignerDb::<D>::completed(txn, batch.block.0) {
+    // Use the batch id as the ID
+    let mut id = [0u8; 32];
+    id.copy_from_slice(&batch.id.to_le_bytes());
```

Comment:
I believe this will panic and you need to scope it as `id[.. 8].copy_from_slice`.

Regardless, we shouldn't use the u64, yet some hash of the batch (similar to how TXs have their IDs created).